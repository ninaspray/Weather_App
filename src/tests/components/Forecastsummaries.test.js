import React from "react";
import { render } from "@testing-library/react";
import ForecastSummaries from "../../components/ForecastSummaries";

describe("ForecastSummaries", () => {
    const validProps = [
        {
            date: 1234,
            description: "stub mock 1",
            icon: "800",
            temperature: {
                max: 30,
                min: 12,
            },
        },
        {
            date: 2468,
            description: "stub mock 2",
            icon: "200",
            temperature: {
                max: 20,
                min: 8,
            },
        },
    ];


it("renders the stub data correctly", () => {
    const { asFragment } = render(<ForecastSummaries forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
});
it("renders correct the ammount of forecasts", () => {
    const { getAllByTestId } = render(
        <ForecastSummaries forecasts={validProps} />
    );
    expect(getAllByTestId("forecast-summary")).toHaveLength(2);
})
});