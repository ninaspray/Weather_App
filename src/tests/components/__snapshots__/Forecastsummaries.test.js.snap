// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastSummaries renders the stub data correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-summaries"
  >
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        1234
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-sunny"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        30°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub mock 1
      </div>
    </div>
    <div
      class="forecast-summary"
      data-testid="forecast-summary"
    >
      <div
        class="forecast-summary__date"
      >
        2468
      </div>
      <div
        class="forecast-summary__icon"
        data-testid="forecast-icon"
      >
        <i
          class="wi wi-day-storm-showers"
          name="owm"
        />
      </div>
      <div
        class="forecast-summary__temperature"
      >
        20°C
      </div>
      <div
        class="forecast-summary__description"
      >
        stub mock 2
      </div>
    </div>
  </div>
</DocumentFragment>
`;
